src/docs/components/preview.svelte

```diff
@@ -0,0 +1,120 @@
+<script lang="ts" context="module">
+	export type PreviewProps = {
+		code: {
+			[key: string]: {
+				'index.svelte'?: string;
+				'tailwind.config.ts'?: string;
+				'globals.css'?: string;
+			} | null;
+		};
+	};
+</script>
+
+<script lang="ts">
+	import { page } from '$app/stores';
+	import { createSelect, type CreateSelectArgs } from '$lib';
+	import { cn } from '$docs/utils';
+	import CodeBlock from './code-block.svelte';
+	import PreviewWrapper from './preview-wrapper.svelte';
+	import { PreviewStyleSelect } from '$docs/components';
+	import Switch from './switch.svelte';
+	import { TabsList, TabsRoot } from './tabs';
+
+	type $$Props = PreviewProps & {
+		viewCode: boolean;
+	};
+
+	export let code: $$Props['code'];
+
+	function handleMissingCode(code: $$Props['code']) {
+		if (!Object.prototype.hasOwnProperty.call(code, 'tailwind')) {
+			code['tailwind'] = null;
+		}
+		if (!Object.prototype.hasOwnProperty.call(code, 'css')) {
+			code['css'] = null;
+		}
+		return code;
+	}
+
+	let codingStyle = Object.keys(handleMissingCode(code))[0]
+		? ('tailwind' as const)
+		: ('css' as const);
+	let codingStyleObj: $$Props['code'][typeof codingStyle] | null =
+		handleMissingCode(code)[codingStyle];
+
+	$: {
+		codingStyleObj = handleMissingCode(code)[codingStyle];
+	}
+
+	$: files = codingStyleObj !== null ? Object.keys(codingStyleObj) : [];
+
+	const { value } = createSelect({
+		value: codingStyle,
+	});
+	value.subscribe((v) => {
+		if (v === 'tailwind' || v === 'css') {
+			codingStyle = v;
+		}
+	});
```

Comment:
I'm confused why we have this and pass it to PreviewStyleSelect. Why not just create a writable and pass it instead?